src/Compilers/CSharp/Portable/Compilation/InitializerSemanticModel.cs

```diff
@@ -275,5 +275,24 @@ protected override void AnalyzeBoundNodeNullability(BoundNode boundRoot, Binder
         {
             NullableWalker.AnalyzeWithoutRewrite(Compilation, MemberSymbol, boundRoot, binder, diagnostics, createSnapshots);
         }
+
+#nullable enable
+
+        protected override bool IsNullableAnalysisEnabled()
+        {
+            switch (MemberSymbol.Kind)
+            {
+                case SymbolKind.Field:
+                case SymbolKind.Property:
+                    return MemberSymbol.ContainingType is SourceMemberContainerTypeSymbol type &&
```

Comment:
Under what conditions would `MemberSymbol.ContainingType is sourceMemberContainerTypeSymbol` be false?

Reply:
Not sure, but it seemed unnecessary to assume the containing type is `SourceMemberContaininerTypeSymbol`.

---
In reply to: [551572303](https://github.com/dotnet/roslyn/pull/49953#discussion_r551572303) [](ancestors = 551572303)

Reply:
I agree it's a bad assumption but my fear was whether or not we were missing a case here: either now or something we should guard against in the future. If this is a standard pattern though think it's fine (couldn't find any supporting examples in the file).